src/multihash.rs

```diff
@@ -47,38 +48,72 @@ pub trait MultihashDigest<C: MultihashCode>: Clone + Debug + Eq + Send + Sync +
     }
 }
 
-/// Trait to compute the digest of a multihash code.
-pub trait MultihashCode:
-    Into<u64> + TryFrom<u64, Error = Error> + Copy + Debug + Eq + Send + Sync + 'static
-{
-    /// Multihash type.
-    type Multihash: MultihashDigest<Self>;
-
-    /// Returns the size of the digest.
-    fn size(&self) -> u8;
-
+/// Trait that makes it possible to create a new hash from some data.
+pub trait MultihashCreate: Clone + Debug + Eq + Send + Sync + 'static {
     /// Returns the hash of the input.
-    fn digest(&self, input: &[u8]) -> Self::Multihash;
+    fn new(code: u64, input: &[u8]) -> Result<Self, Error>;
```

Comment:
Error type should probably be `UnsupportedCode`. That way most applications can just unwrap here.

Reply:
I don't understand your comment. The implementation currently returns an `Error::UnsupportedCode()`. Is there a way to specify more strictly in the return type?

Reply:
Create a second error type

Reply:
I don't have enough experience with library APIs. So far I've mostly encountered one error type per library. But we can certainly introduce a new error type if that makes the API nicer.

Reply:
Well if the only possible error is UnsupportedCode, but you return an enum, it's not clear that it's safe to unwrap without reading the code. If the error type is `struct UnsupportedCode`, it's clear that it can be unwrapped. The error enum can include the unsupported code error as #[from] UnsupportedCode, if it's necessary

Reply:
I guess it's not that important, do you want to merge?

Reply:
Thanks for the elaborate reply, that makes sense now. Yes I'll merge.